### ISSUE TITLE
`esm.sh` incorrectly transforms Solid JSX

### ISSUE THREAD

**Issue by LePichu** [ije/esm.sh#496]
# Introduction 
As Solid is far different than traditional JSX transformation, `esm.sh` produces very incorrect and React oriented JSX which can not be used in Solid even tho it's importing from Solid.

# Result
An example of incorrect transformation: https://esm.sh/v102/@solidjs/router@0.6.0/deno/dist/index.jsx.development.js

# Expected 
What should happen is either it does not transform and return JSX or return an error that this module is not supported.

**Comment by ije** [comment]
if we just return the raw jsx, can it work in deno runtime?

**Comment by LePichu** [comment]
> if we just return the raw jsx, can it work in deno runtime?

Deno can deal with JSX/TSX, but also the fact that most people will be using something like ESBuild and that will take care of building the final output from JSX.

**Comment by LePichu** [comment]
![IMG_20230112_022802](https://user-images.githubusercontent.com/62379502/211916454-e5446838-41b7-4d67-8074-62675ab0c9f4.jpg)
Reconfirmation; but JSX works/runs out of box on Deno.

**Comment by ije** [comment]
Other question, how do you get the url https://esm.sh/v102/@solidjs/router@0.6.0/deno/dist/index.jsx.development.js ?

<img width="677" alt="Screenshot 2023-01-12 at 04 58 42" src="https://user-images.githubusercontent.com/2883484/211916401-e8b6fbf4-ed03-4a1f-a60d-1eb1da1d9cff.png">

**Comment by ije** [comment]
> ![IMG_20230112_022802](https://user-images.githubusercontent.com/62379502/211916454-e5446838-41b7-4d67-8074-62675ab0c9f4.jpg) Reconfirmation; but JSX works/runs out of box on Deno.

yeah i know deno supports jsx, but seems solidjs is using babel to do ATO transplile

**Comment by LePichu** [comment]
> Other question, how do you get the url https://esm.sh/v102/@solidjs/router@0.6.0/deno/dist/index.jsx.development.js ?
> 
> <img alt="Screenshot 2023-01-12 at 04 58 42" width="677" src="https://user-images.githubusercontent.com/2883484/211916401-e8b6fbf4-ed03-4a1f-a60d-1eb1da1d9cff.png">

It uhh.....

![Screenshot_2023-01-12-02-33-30-47_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/62379502/211917369-4d34bed5-90bc-4f14-8c65-331e9e502d3d.jpg)

**Comment by LePichu** [comment]
> > ![IMG_20230112_022802](https://user-images.githubusercontent.com/62379502/211916454-e5446838-41b7-4d67-8074-62675ab0c9f4.jpg) Reconfirmation; but JSX works/runs out of box on Deno.
> 
> yeah i know deno supports jsx, but seems solidjs is using babel to do ATO transplile

Is it not possible to return Raw JSX? Just transpiling the TS part and keeping the JSX intact? There's an ESBuild plugin to handle Solid JSX Transforms tho which we use, not sure if that's helpful.

**Comment by ije** [comment]
in fact you can get the raw jsx code via https://esm.sh/@solidjs/router@0.6.0/dist/index.jsx

**Comment by LePichu** [comment]
> > There's an ESBuild plugin to handle Solid JSX Transforms tho which we use, not sure if that's helpful.
> 
> i think the plugin also uses babel too

It does.

**Comment by LePichu** [comment]
> in fact you can get the raw jsx code via https://esm.sh/@solidjs/router@0.6.0/dist/index.jsx

I'll try around with this, thanks!

**Comment by Suyashtnt** [comment]
> in fact you can get the raw jsx code via [esm.sh/@solidjs/router@0.6.0/dist/index.jsx](https://esm.sh/@solidjs/router@0.6.0/dist/index.jsx)

This in deno doesn't return the raw JSX, but rather redirects to a precompiled version which uses `React` as a way to generate components

**Comment by LePichu** [comment]
@ije I think having something like allowing to use ESBuild plugins in `esm.sh` to transform JS flavors properly could work here and also fix issues with other frameworks if there are any.
